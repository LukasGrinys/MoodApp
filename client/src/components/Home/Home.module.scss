@import '../../styles/config/variables';

.wrapper {
    text-align: center;

    .bigLogo {
        font-size: 4.5em;
        color: $color-primary;
        font-family: "Kaushan Script";
        text-align: center;
        margin-bottom:25px;

        &.dark {
            color: $color-night-white;
        }
    }
    
    .text {
        font-weight: 300;
        padding: 25px;
        text-align: center;
        font-size: 1.5em;
        margin-bottom: 25px;

        & > span {
            color: $color-primary;
        }
    }

    .buttonWrapper {
        display: flex;
        flex-direction: row;
        align-items: normal;
        justify-content: space-around;
        flex-wrap: wrap;
    }
}