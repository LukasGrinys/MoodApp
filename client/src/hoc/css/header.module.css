.header {
    color: #f2f2f2;
    width: 100%;
    position: sticky;
    top: 0;
    left: 0;
    display: flex;
    flex-direction: column;
    z-index:996;
}
.top {
    position: relative;
    background-color: #3366ff;
    box-shadow: 2px 2px 12px rgba(0,0,0,0.3);
    display: flex;
    flex-direction: row;
    padding: 8px;
    z-index: 999;
}
.logo {
    position: absolute;
    top: 4px;
    font-family: "Kaushan Script";
    font-size: 1.4em;
    cursor: default;
}
.bars {
    margin: auto;
    cursor:pointer;
}
.navBar {
    position: absolute;
    top: 44px;
    left: 0;
    display: flex;
    flex-direction: column;
    width: 100%;
    z-index: 998;
    animation: appearTop 0.5s;
}
.navBarDark {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0,0,0,0.5);
    z-index: 997;
    animation: fadeIn 0.5s;
}
.navItem {
    width: 100%;
    text-align: left;
    padding: 6px;
    background-color: #e0ebeb;
    color: #141f1f;
    cursor: pointer;
    z-index: 998;
}
.navItem:hover {
    color: #3366ff;
}
.bars {
    animation: rotateButton 2s;
    transition: 1s;
}
a {
    text-decoration: none;
    color: black;
}

@keyframes fadeIn {
    from { opacity: 0%;}
    to { opacity: 100%;}
}

@keyframes appearTop {
    from { 
        transform: translateY(-100%);
        opacity: 0;
    }
    to {
        transform: translateY(0%);
        opacity: 1;
    }
}