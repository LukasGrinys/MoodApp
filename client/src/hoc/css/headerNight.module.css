.header {
    color: #f2f2f2;
}
.top {
    background-color: #2F2F2F;
}