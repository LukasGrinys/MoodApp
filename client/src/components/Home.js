import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import ScrollAnimation from 'react-animate-on-scroll';
import "animate.css/animate.min.css";

import styles from './home.module.css';
import ButtonWid from './../widgets/Button';
import Loading from '../widgets/loading';

const DefaultHome = () => {
    return (
        <div className={styles.wrapper} >
            <ScrollAnimation animateIn="fadeInDown" animateOnce={true}>
                <div className={styles.bigLogo}>
                    MoodApp
                </div>
            </ScrollAnimation>
            <ScrollAnimation animateIn="fadeInDown" delay={1000} offset={75} animateOnce={true}>
                <div className={styles.text}>
                    Your daily mood tracker
                </div>
            </ScrollAnimation>
            
            <ScrollAnimation animateIn="fadeInDown" delay={2000} offset={75} animateOnce={true}>
                <div className={styles.buttons}>
                    <Link to="/login">
                        <ButtonWid color={"#3366ff"} background={"#fff2f2"}>
                            Log In
                        </ButtonWid>
                    </Link>
                    <Link to="/register">
                        <ButtonWid color={"#fffff2"} background={"#3366ff"}>
                            Sign Up
                        </ButtonWid>
                    </Link>
                </div>
            </ScrollAnimation> 
        </div>
    )
}

const UserHome = (props) => {
    return (
        <div className={styles.wrapper} >
            <ScrollAnimation animateIn="fadeInDown" animateOnce={true}>
                <div className={styles.bigLogo}>
                    MoodApp
                </div>
            </ScrollAnimation>
            <ScrollAnimation animateIn="fadeInDown" delay={1000} offset={75} animateOnce={true}>
                <div className={styles.textAlt}>
                    Hello <span style={{color:"#3366ff"}}>{props.name}</span>
                </div>
            </ScrollAnimation>
            
            <ScrollAnimation animateIn="fadeInDown" delay={2000} offset={75} animateOnce={true}>
                <div className={styles.buttons}>
                    <Link to="/dashboard">
                        <ButtonWid color={"#fffff2"} background={"#3366ff"}>
                            Your Dashboard
                        </ButtonWid>
                    </Link>
                    <Link to="/logout">
                        <ButtonWid color={"#3366ff"} background={"#fff2f2"}>
                            Log Out
                        </ButtonWid>
                    </Link>
                </div>
            </ScrollAnimation> 
        </div>
    )
}


const Home = ({user, nightmode}) => {
    if (user.data) {
        if (user.data.isAuth) {
            return (<UserHome name={user.data.firstName} nightmode={nightmode}/>)
        }
        if (!user.data.isAuth) {
            return (<DefaultHome nightmode={nightmode}/>)
        }
    } else {
        return ( <Loading nightmode={nightmode}/>)
    }
};

function mapStateToProps(state) {
    return {
        user: state.user
    }
}
export default connect(mapStateToProps)(Home);