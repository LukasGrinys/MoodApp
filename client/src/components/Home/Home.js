import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import ScrollAnimation from 'react-animate-on-scroll';
import "animate.css/animate.min.css";
import classNames from 'classnames';
import styles from './Home.module.scss';
import Button from '../Button/Button';
import Loading from '../Loading/loading';
import { useTheme } from '../../contexts/ThemeContext';

const Home = ({isAuth, firstName}) => {    
    const darkTheme = useTheme();

    if (isAuth === undefined) {
        return <Loading/>
    }

    return  (
    <div className={styles.wrapper}>
        <ScrollAnimation animateIn="fadeInDown" animateOnce={true}>
            <div className={classNames(styles.bigLogo, darkTheme && styles.dark)}>
                MoodApp
            </div>
        </ScrollAnimation>
        <ScrollAnimation animateIn="fadeInDown" delay={1000} offset={75} animateOnce={true}>
            {   isAuth ? (
                    <div className={styles.text}>
                        Hello <span>{firstName}</span>  
                    </div>
                ) : ( <div className={styles.text}>Your daily mood tracker</div> )
            }
        </ScrollAnimation>
        <ScrollAnimation animateIn="fadeInDown" delay={2000} offset={75} animateOnce={true}>
            <div className={styles.buttonWrapper}>
                {
                    isAuth ? (
                        <Fragment>
                            <Link to="/dashboard">
                                <Button color={'primary'}>
                                    Dashboard
                                </Button>
                            </Link>
                            <Link to="/logout">
                                <Button color={'white'}>
                                    Log out
                                </Button>
                            </Link>
                        </Fragment>
                    ) : (
                        <Fragment>
                            <Link to="/login">
                                <Button color={'primary'}>Log in</Button>
                            </Link>
                            <Link to="/register">
                                <Button color={'white'}>Register</Button>
                            </Link>
                        </Fragment>
                    )
                }
            </div>
        </ScrollAnimation> 
    </div>)
};

const mapStateToProps = ({user}) => {
    const { isAuth, userData } = user;
    const firstName = userData ? userData.firstName : '';

    return {
        isAuth,
        firstName
    }
}
export default connect(mapStateToProps)(Home);