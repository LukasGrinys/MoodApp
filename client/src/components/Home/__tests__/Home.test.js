import configureStore from 'redux-mock-store';
import React from 'react';
import "animate.css/animate.min.css";
import Home from '../Home';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
configure({ adapter: new Adapter() });
const mockStore = configureStore();

jest.mock('react-animate-on-scroll');

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect,
  }));

test('renders loading screen if there is no auth data', () => {
    const state = {
        user : {}
    };
    const store = mockStore(state);

    const wrapper = shallow(<Home store={store}/>);
    expect(wrapper).toMatchSnapshot();
});


test('renders page if there is user auth data', () => {
    const state = {
        user : {
            isAuth : false
        }
    };
    const store = mockStore(state);

    const wrapper = mount(<Home store={store}/>);
    expect(wrapper).toMatchSnapshot();
});