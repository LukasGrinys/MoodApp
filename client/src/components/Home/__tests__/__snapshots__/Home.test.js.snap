// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading screen if there is no auth data 1`] = `
<ContextProvider
  value={null}
>
  <Home
    dispatch={[Function]}
    firstName=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`renders page if there is user auth data 1`] = `
<Connect(Home)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Home
    dispatch={[Function]}
    firstName=""
    isAuth={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div
      className="wrapper"
    >
      <ScrollAnimation
        animateIn="fadeInDown"
        animateOnce={true}
        animatePreScroll={true}
        delay={0}
        duration={1}
        initiallyVisible={false}
        offset={150}
      />
      <ScrollAnimation
        animateIn="fadeInDown"
        animateOnce={true}
        animatePreScroll={true}
        delay={1000}
        duration={1}
        initiallyVisible={false}
        offset={75}
      />
      <ScrollAnimation
        animateIn="fadeInDown"
        animateOnce={true}
        animatePreScroll={true}
        delay={2000}
        duration={1}
        initiallyVisible={false}
        offset={75}
      />
    </div>
  </Home>
</Connect(Home)>
`;
